streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -217,7 +217,7 @@ private String ensureOneNodeIsReady(final List<Node> nodes) {
                 }
             }
             try {
-                kafkaClient.poll(500, Time.SYSTEM.milliseconds());
+                kafkaClient.poll(0, Time.SYSTEM.milliseconds());
```

Comment:
Yes. Zero is not blocking. We just need to call `poll` trigger an update for `ready()` call from above (cf. JavaDoc comment of `ready`).